### PR TITLE
Support short topic name in cpp client

### PR DESCRIPTION
**Pull request by sijie**
This related to #1535 - support short topic name in cpp client.

### CODE REVIEW

File: pulsar-client-cpp/lib/TopicName.cc [hunk 1]
```diff
@@ -52,8 +52,19 @@ TopicName::TopicName() {}
 bool TopicName::init(const std::string& topicName) {
     topicName_ = topicName;
     if (topicName.find("://") == std::string::npos) {
-        LOG_ERROR("Topic name is not valid, domain not present - " << topicName);
-        return false;
+        std::string topicNameCopy_ = topicName;
+        std::vector<std::string> pathTokens;
+        boost::algorithm::split(pathTokens, topicNameCopy_, boost::algorithm::is_any_of("/"));
+        if (pathTokens.size() == 3) {
+          topicName_ = "persistent://" + pathTokens[0] + "/" + pathTokens[1] + "/" + pathTokens[2];
```
Comment (merlimat) [inline pulsar-client-cpp/lib/TopicName.cc]: This will probably break the format checker that expects 4 spaces indent

Comment (sijie) [reply pulsar-client-cpp/lib/TopicName.cc]: nice catch.

**Comment by sijie** [comment]
@merlimat fixed the tests

**Comment by merlimat** [comment]
@sijie I think the c++ format check failed to validate.

**Comment by sijie** [comment]
retest this please

**Pull request merged**